tinytuya/core.py

```diff
@@ -88,18 +88,37 @@
 except NameError:
     pass
 
-# Required module: pycryptodome
-try:
-    import Crypto
-    from Crypto.Cipher import AES  # PyCrypto
-except ImportError:
-    Crypto = AES = None
-    import pyaes  # https://github.com/ricmoo/pyaes
+for clib in ('pyca/cryptography', 'PyCryptodomex', 'PyCrypto', 'pyaes'):
+    Crypto = AES = CRYPTOLIB = None
+    try:
+        if clib == 'pyca/cryptography': # https://cryptography.io/en/latest/
+            from cryptography.hazmat.primitives.ciphers import Cipher as Crypto
+            from cryptography.hazmat.primitives.ciphers import modes as Crypto_modes
+            from cryptography.hazmat.primitives.ciphers.algorithms import AES
+            from cryptography import __version__ as Crypto_version
+        elif clib == 'PyCryptodomex': # https://pycryptodome.readthedocs.io/en/latest/
+            # PyCryptodome is installed as "Cryptodome" when installed by
+            #  `apt install python3-pycryptodome` or `pip install pycryptodomex`
+            import Cryptodome as Crypto
+            from Cryptodome.Cipher import AES
+        elif clib == 'PyCrypto': # https://www.pycrypto.org/
+            import Crypto
+            from Crypto.Cipher import AES
+            # v1/v2 is PyCrypto, v3 is PyCryptodome
+            clib = 'PyCrypto' if Crypto.version_info[0] < 3 else 'PyCryptodome'
+        elif clib == 'pyaes':
+            import pyaes  # https://github.com/ricmoo/pyaes
+        CRYPTOLIB = clib
+        break
+    except ImportError:
+        continue
+if CRYPTOLIB is None:
+    raise ModuleNotFoundError('No crypto library found, please install one of: pyca/cryptography, PyCryptodome, or PyCrypto')

```

Comment:
Should we just message this as the `pip install` equivalent?

No crypto library found, please install: cryptography, pycryptodome, or pyaes

Reply:
Nevermind, I'm going to leave it as is.  This is similar to README.

Reply:
Too late...